### ISSUE TITLE
error: unstable feature

### ISSUE THREAD

**Issue by OlivierDeRivoyre** [sondrele/rust-bmp#18]
Hi Sondre,

It's my first try on Rust, after doing an "Hello world" I've try to use your lib: I'm really a noob, it may be a problem on my side, not yours.

Having "rustc 1.0.0-beta.3", I've done as on your example. I've got this error:
```
> cargo build
   Compiling bmp v0.1.2 (https://github.com/sondrele/rust-bmp#060fb24d)
C:\Users\Yop\.cargo\git\checkouts\rust-bmp-3a7f890ae607b0ba\master\src\lib.rs:3:1: 3:25 error: unstable feature
C:\Users\Yop\.cargo\git\checkouts\rust-bmp-3a7f890ae607b0ba\master\src\lib.rs:3 #![feature(collections)]
                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~
note: this feature may not be used in the beta release channel
error: aborting due to previous error
Could not compile `bmp`.

To learn more, run the command again with --verbose.
```

**Comment by sondrele** [comment]
Hi!

Awesome that you went straight to this little library! :)

The reason you're getting this error is because you have Rust beta, and the beta does not support unstable features. 
This library uses the `collections` crate from Rust itself, but parts of `collections` is unfortunately still marked as unstable, and there's the issue.

I wonder, does it work if you put the following into your Cargo.toml, instead of as a git dependency?
```toml
[dependencies]
bmp = "*"
```

Another thing you can do is also to download and install [Rust nightly](http://www.rust-lang.org/install.html) (at the bottom of this page). The nightly does not break with `unstable` features.

**Comment by OlivierDeRivoyre** [comment]
Hi Sondre

After installing the Rust nightly, it works!

I've then got a strange 'arithmetic operation overflowed' panic while trying your example. With a small change, it was running:

```
#[macro_use]
extern crate bmp;
use bmp::{Image, Pixel};

fn main() {
    let mut img = Image::new(256, 256);
    for (x, y) in img.coordinates() {
        img.set_pixel(x, y, px!(x, y,200));
        //instead of img.set_pixel(x, y, px!(x - y / 255, y - x / 255, x + y / 255));	   
    }
    let _ = img.save("img.bmp");
}
```

From my point of view, it is easy to understand why your lib is useful and will probably be used a lot in the near future by other novices.
After reading some tutorial, I wanted a cool idea of project to practice. Since Rust is as low level as C, it should have the perf to play with pixel in a bitmap. While handling a window probably ask some knowledge, it will be easy to set pixels in a bitmap and save the bitmap in a file to see the result. A search on google for "Rust Bitmap" and here you are with your perfect readme :)

Thanks for your quick answer (and for your lib of course)!

**Comment by sondrele** [comment]
Cool!

Same story for me last year when I started learning Rust! If I remember correctly, there were no image libraries around at that time, so I started implemented this small piece.

I actually decided to quit relying on the collections crate when you brought this to my notice yesterday, so I cleaned up some parts of the library today, it should be compatible with the beta now :)
I'll update it with a new version shortly.

Also, even though the Rust developers recommend the beta (for beginners at least), I've found that the nightly is still a better option. It's still really stable, and there's a lot of packages out there that still rely on some unstable feature from different libraries. 

And, I'll update the readme with an example that compiles properly, thanks!
